service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -222,145 +230,131 @@ private boolean updateWorkspaceColumns(UUID workspaceId, MapSqlParameterSource c
    */
   @ReadTransaction
   public Optional<GcpCloudContext> getGcpCloudContext(UUID workspaceId) {
-    String sql =
-        "SELECT context FROM cloud_context "
-            + "WHERE workspace_id = :workspace_id AND cloud_platform = :cloud_platform";
-    MapSqlParameterSource params =
-        new MapSqlParameterSource()
-            .addValue("workspace_id", workspaceId.toString())
-            .addValue("cloud_platform", CloudPlatform.GCP.toString());
-    try {
-      return Optional.ofNullable(
-          DataAccessUtils.singleResult(
-              jdbcTemplate.query(
-                  sql,
-                  params,
-                  (rs, rowNum) -> deserializeGcpCloudContext(rs.getString("context")))));
-    } catch (EmptyResultDataAccessException e) {
-      return Optional.empty();
-    }
+    return getCloudContextWorker(workspaceId, CloudPlatform.GCP).map(GcpCloudContext::deserialize);
   }
 
   /**
-   * Create the cloud context of the workspace
+   * Create the GCP cloud context of the workspace
    *
    * @param workspaceId unique id of the workspace
    * @param cloudContext the GCP cloud context to create
    */
   @WriteTransaction
-  public void createGcpCloudContext(UUID workspaceId, GcpCloudContext cloudContext) {
+  public void createGcpCloudContext(
+      UUID workspaceId, GcpCloudContext cloudContext, String flightId) {
+    createCloudContextWorker(workspaceId, CloudPlatform.GCP, cloudContext.serialize(), flightId);
+  }
+
+  /**
+   * Delete the GCP cloud context for a workspace
+   *
+   * @param workspaceId workspace of the cloud context
+   */
+  @WriteTransaction
+  public void deleteGcpCloudContext(UUID workspaceId) {
```

Comment:
worth a comment comparing this to the other deleteGcpCloudContext method to explain when this is appropriate to use